numpy/lib/arraypad.py

```diff
@@ -53,24 +53,29 @@ def _linear_ramp(ndim, axis, start, stop, size, reverse=False):
 
     Examples
     --------
-    >>> _linear_ramp(ndim=2, axis=0, start=np.arange(3), stop=10, size=2)
+    >>> _linear_ramp(ndim=2, axis=0, start=np.arange(3), stop=10, num=2)
     array([[0. , 1. , 2. ],
            [5. , 5.5, 6. ]])
-    >>> _linear_ramp(ndim=3, axis=0, start=2, stop=0, size=0)
+    >>> _linear_ramp(ndim=3, axis=0, start=2, stop=0, num=0)
     array([], shape=(0, 1, 1), dtype=float64)
     """
+    # Use smallest signed dtype that can accommodate true division
+    dtype = np.result_type(np.float16, start, stop)
```

Comment:
Thanks for the ping chuck. The `np.float16` seems still a bit strange to me. Can we make this `result_type` include final output? That way you get high precision floating point if necessary. I realize that you do not have that dtype available here naturally. An alternative may be to cast the input values early on as well.

Of course that might still fail if everything is unsigned, hmmm.

The current `np.float16` is not trustworthy, a scalar float could be better. This way if start and stop are scalars, the result dtype will actually end up being float16, and we do not want that happening.

Reply:
What do you mean by "final output"? In the current implementation the parameter `stop` is the slice of the edge of the padded array and has the same dtype as the returned array.

What do you mean by "scalar float"? This?
```python
np.result_type(float, start, stop)
```

Reply:
Oh, I mean `np.result_type(start, stop, 0.)`, final output is the input array dtype, yes. I think it should rather be `np.result_type(start, stop, input_dtype)` maybe, although if all are uint, things are different.

Reply:
Didn't notice the edge cases between `float` and `0.` as parameters. I think `np.result_type(start, stop, 0.)` fulfills all our constraints.

Thought: Maybe we should move the statement `_round_if_needed(right_ramp, padded.dtype)` (called right after this function) into it. That would consolidate the dtype management into one place.

Reply:
Can we just call `linspace` here, and leave it to handle the corner cases?

Reply:
Huh, linspace supports non-scalars since 1.16. Good to know! Your suggestion doesn't work because the reshaping doesn't translate to the output of linspace. However I'll see how I can use linspace here and how it compares.

Reply:
Found a solution and moved that one into _get_linear_ramps and deleted the obsolete _linear_ramp function. Haven't yet benchmarked it.